osquery/tables/applications/darwin/browser_plugins.cpp

```diff
@@ -227,26 +281,208 @@ QueryData genSafariExtensions(QueryContext& context) {
       continue;
     }
 
-    auto dir = fs::path(directory->second) / kSafariExtensionsPath;
+    auto dir = fs::path(directory->second) / kSafariLegacyExtensionsPath;
     // Check that an extensions directory exists.
     if (!pathExists(dir).ok()) {
       continue;
     }
 
     // Glob the extension files.
     std::vector<std::string> paths;
-    if (!resolveFilePattern(
-             dir / kSafariExtensionsPattern, paths, GLOB_ALL | GLOB_NO_CANON)
+    if (!resolveFilePattern(dir / kSafariLegacyExtensionsPattern,
+                            paths,
+                            GLOB_ALL | GLOB_NO_CANON)
              .ok()) {
       continue;
     }
 
     for (const auto& extension_path : paths) {
-      genSafariExtension(uid->second, gid->second, extension_path, results);
+      genSafariLegacyExtension(
+          uid->second, gid->second, extension_path, results);
     }
   }
+}
 
-  return results;
+inline bool getPtreeFromPlist(const fs::path& plist_path, pt::ptree& tree) {
+  // Perform a dry run of the file read.
+  if (!isReadable(plist_path).ok()) {
+    return false;
+  }
+
+  // Read the plist file
+  std::string content;
+  if (!readFile(plist_path, content).ok()) {
+    return false;
+  }
+
+  // Parse the plist content
+  if (!parsePlistContent(content, tree).ok()) {
+    return false;
+  }
+
+  return true;
+}
+
+inline void getSandboxedExtensionData(const std::string& extension_path,
+                                      const std::string& plist_path,
+                                      SandboxedExtensionsData& data) {
+  // Parse the plist content
+  pt::ptree tree;
+  if (!getPtreeFromPlist(plist_path, tree)) {
+    return;
+  }
+
+  // Populating SandboxedExtensionData entry
+  SandboxedExtensionData entry;
+  entry.app_path = extension_path;
+  entry.identifier = tree.get<std::string>("CFBundleIdentifier", "");
+  entry.name = tree.get<std::string>("CFBundleDisplayName", "");
+  entry.sdk = tree.get<std::string>("CFBundleInfoDictionaryVersion", "");
+  entry.version = tree.get<std::string>("CFBundleShortVersionString", "");
+  data.push_back(entry);
+}
+
+inline bool isExtensionAppExcluded(const std::string& app_dir) {
+  if (app_dir.empty()) {
+    return false;
+  }
+
+  for (const auto& app_to_exclude : kExtensionsAppDirsToExclude) {
+    if (boost::algorithm::contains(app_dir, app_to_exclude)) {
+      return true;
+    }
+  }
+
+  return false;
 }
+
+inline bool isUserExtension(const fs::path& app_extension_plist,
+                            const fs::path web_extension_plist,
+                            const SandboxedExtensionData& ext_data) {
+  // Gather user extension metainformation
+  pt::ptree app_extension_ptree;
+  pt::ptree web_extension_ptree;
+  getPtreeFromPlist(app_extension_plist, app_extension_ptree);
+  getPtreeFromPlist(web_extension_plist, web_extension_ptree);
```

Comment:
I'm just walking around and asking some questions ;)
Will it make some sense to populate `web_extension_ptree` right after the for cycle over `app_extension_ptree`?
What if the function returns on line 372 and call for `getPtreeFromPlist(web_extension_ptree)` will not be used?